app/app.go

```diff
@@ -0,0 +1,107 @@
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/app"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/store/iavl"
+	"github.com/iov-one/weave/x"
+	"github.com/iov-one/weave/x/msgfee"
+	"github.com/iov-one/weave/x/multisig"
+	"github.com/iov-one/weave/x/sigs"
+	"github.com/iov-one/weave/x/utils"
+)
+
+// Authenticator returns authentication with multisigs
+// and public key signatues
+func Authenticator() x.Authenticator {
+	return x.ChainAuth(sigs.Authenticate{}, multisig.Authenticate{})
+}
+
+// Chain returns a chain of decorators, to handle authentication,
+// fees, logging, and recovery
+func Chain(authFn x.Authenticator, minFee coin.Coin) app.Decorators {
+
+	// TODO implement orderbook controller
+	return app.ChainDecorators(
+		utils.NewLogging(),
+		utils.NewRecovery(),
+		utils.NewKeyTagger(),
+		utils.NewSavepoint().OnCheck(),
+		sigs.NewDecorator(),
+		multisig.NewDecorator(authFn),
+		utils.NewSavepoint().OnDeliver(),
+		msgfee.NewFeeDecorator(),
```

Comment:
msgfee can probably be removed from base app.
And the cash.StaticFeeDecorator can be used above (this is the simplest case when the product fee doesn't depend on the output from the Handler)